src/lib/utilities/decode-payload.ts

```diff
@@ -83,27 +83,42 @@ export const decodePayloadAttributes = (
 };
 
 // List of fields with payloads
-const payloadFields: string[] = [
-  'input',
-  'result',
-  'details',
-  'heartbeatDetails',
-  'lastHeartbeatDetails',
-  'lastFailure',
-  'lastCompletionResult',
-  'queryArgs',
-  'answer',
-  'signalInput',
+const payloadFields: string[][] = [
+  ['data'],
+  ['input'],
+  ['result'],
+  ['details', 'change-id'],
+  ['details', 'data'],
+  ['details', 'result'],
+  ['details', 'version'],
+  ['details'],
+  ['heartbeatDetails'],
+  ['lastHeartbeatDetails'],
+  ['lastFailure'],
+  ['lastCompletionResult'],
+  ['queryArgs'],
+  ['answer'],
+  ['signalInput'],
 ];
```

Comment:
This is a tiny little nit that you can ignore, but this _might_ make the TypeScript typings better and then you can drop the `string[][]` on L86.

```suggestion
] as const;
```